### PR TITLE
Add getting started section

### PR DESCRIPTION
**Pull request by hickeyma**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,21 @@ Caikit provides an abstraction layer for application developers where they can c
...
+
+The model can then be served by using gRPC, Knative or KServe. Here is an example of the [text sentiment server code for gRPC](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/start_runtime.py). This references the module configuration [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/models/text_sentiment/config.yml). This configuration specifies the module(s) (which wraps the model(s)) to serve.
+
+There is an example of a clien [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/client.py) which is a simple Python CLI which calls the model and queries it for sentiment analysis on 2 different pieces of text. The client also references the module configuration.
```
Comment (markstur) [inline README.md:36]: typo "clien"

Comment (hickeyma) [reply README.md]: Nice catch. Fixed.

File: README.md [hunk 2]
```diff
@@ -28,6 +28,21 @@ Caikit provides an abstraction layer for application developers where they can c
...
 
+## Getting Started
+
+There are 2 key things to define upfront when using Caikit to manage your AI model. They are as follows:
+
+- [Module](https://github.com/caikit/caikit/blob/main/docs/adrs/001-module.md)
+- [Data model](https://github.com/caikit/caikit/blob/main/docs/adrs/010-data-model-definition.md)
+
+The `module` defines the entry points for Caikit to manage your model. In other words, it tells Caikit how to load, infer and train your model. An example is the [text sentiment module](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/text_sentiment/runtime_model/hf_module.py). The `data model` defines the input and outputs of the model task. An example is the [text sentiment data model](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/text_sentiment/data_model/classification.py).
```
Comment (gabe-l-hart) [inline README.md:38]: I think we'll want to refine the language about the `data model` here once we've got the `caikit.interfaces` matured to include most of the standard `@task` and `@dataobject` definitions. For now, this is good as is.

File: README.md [hunk 3]
```diff
@@ -28,6 +28,21 @@ Caikit provides an abstraction layer for application developers where they can c
...
+
+The model can then be served by using gRPC, Knative or KServe. Here is an example of the [text sentiment server code for gRPC](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/start_runtime.py). This references the module configuration [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/models/text_sentiment/config.yml). This configuration specifies the module(s) (which wraps the model(s)) to serve.
```
Comment (gabe-l-hart) [inline README.md:40]:
```suggestion
The model can then be served with a gRPC interface using the container runtime of your choice, including Knative and KServe. Here is an example of the [text sentiment server code for gRPC](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/start_runtime.py). This references the module configuration [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/models/text_sentiment/config.yml). This configuration specifies the module(s) (which wraps the model(s)) to serve.
```

Comment (gabe-l-hart) [reply README.md:40]: I think it's important to indicate that `gRPC` is the interface and that `knative` and `kserve` are _examples_ of how to get it off the ground, but are certainly not the only way to run it

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
This is really nice to get in, thanks Martin! A couple minor language tweaks and a reminder that the discussion of the `data model` will be due for some updates once we've matured `caikit.interfaces` (nothing to change there for now, though).

File: README.md (hunk 2 above)
Comment (hickeyma) [reply README.md:38]: ok, noted

File: README.md (hunk 3 above)
Comment (hickeyma) [reply README.md:40]: Good catch.  Updated.

**Comment by hickeyma** [comment]
Thanks for the review and feedback @gabe-l-hart. Updated and ready for review again.

File: README.md [hunk 4]
```diff
@@ -28,6 +28,21 @@ Caikit provides an abstraction layer for application developers where they can c
...
+
+The `module` defines the entry points for Caikit to manage your model. In other words, it tells Caikit how to load, infer and train your model. An example is the [text sentiment module](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/text_sentiment/runtime_model/hf_module.py). The `data model` defines the input and outputs of the model task. An example is the [text sentiment data model](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/text_sentiment/data_model/classification.py).
+
+The model is served by a [gRPC](https://grpc.io) server which can run as is or in container runtime, including [Knative](https://knative.dev/docs/) and [KServe](https://www.kubeflow.org/docs/external-add-ons/kserve/kserve/). Here is an example of the [text sentiment server code for gRPC](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/start_runtime.py). This references the module configuration [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/models/text_sentiment/config.yml). This configuration specifies the module(s) (which wraps the model(s)) to serve.
```
Comment (gabe-l-hart) [inline README.md:40]:
```suggestion
The model is served by a [gRPC](https://grpc.io) server which can run as is or in any container runtime, including [Knative](https://knative.dev/docs/) and [KServe](https://www.kubeflow.org/docs/external-add-ons/kserve/kserve/). Here is an example of the [text sentiment server code for gRPC](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/start_runtime.py). This references the module configuration [here](https://github.com/caikit/caikit/blob/main/examples/text-sentiment/models/text_sentiment/config.yml). This configuration specifies the module(s) (which wraps the model(s)) to serve.
```

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
Minor language tweak

**Review by gabe-l-hart — APPROVED** [review]

**Pull request merged**